src/internals/object.rs

```diff
@@ -0,0 +1,151 @@
+                }
+                yara_sys::OBJECT_TYPE_FLOAT => {
+                    if self.0.value.i == yara_sys::YR_UNDEFINED {
+                        YrObjectValue::Undefined
+                    } else {
+                        YrObjectValue::Float(self.0.value.d)
+                    }
+                }
+                yara_sys::OBJECT_TYPE_STRING => {
+                    let p = self.0.value.ss;
+                    if p.is_null() {
+                        YrObjectValue::Undefined
+                    } else {
+                        YrObjectValue::String(std::slice::from_raw_parts(
+                            (*p).c_string.as_ptr().cast(),
+                            (*p).length as usize,
+                        ))
+                    }
+                }
+                yara_sys::OBJECT_TYPE_STRUCTURE => {
+                    let this: &yara_sys::YR_OBJECT_STRUCTURE = std::mem::transmute(self.0);
+                    let mut members = Vec::new();
+
+                    let mut member = this.members;
+                    while !member.is_null() {
+                        let obj = (*member).object;
+                        if !obj.is_null() {
+                            members.push(YrObject::from(&*obj));
+                        }
+                        member = (*member).next;
+                    }
+                    YrObjectValue::Structure(members)
+                }
+                yara_sys::OBJECT_TYPE_ARRAY => {
+                    let this: &yara_sys::YR_OBJECT_ARRAY = std::mem::transmute(self.0);
+                    if this.items.is_null() {
+                        return YrObjectValue::Array(Vec::new());
+                    }
+
+                    let objects: &[*const yara_sys::YR_OBJECT] = std::slice::from_raw_parts(
+                        (*this.items).objects.as_ptr().cast(),
+                        (*this.items).length as usize,
+                    );
+
+                    YrObjectValue::Array(
+                        objects
+                            .iter()
+                            .map(|v| {
+                                if v.is_null() {
+                                    None
+                                } else {
+                                    Some(YrObject::from(&**v))
+                                }
+                            })
+                            .collect(),
+                    )
+                }
+                yara_sys::OBJECT_TYPE_DICTIONARY => {
+                    let this: &yara_sys::YR_OBJECT_DICTIONARY = std::mem::transmute(self.0);
+                    if this.items.is_null() {
+                        return YrObjectValue::Dictionary(HashMap::new());
+                    }
+
+                    let len = (*this.items).used;
+
+                    YrObjectValue::Dictionary(
+                        (0..len)
+                            .filter_map(|i| {
+                                let v = (*this.items).objects[i as usize];
```

Comment:
There is an issue here, objects is declared with length 1 in "YR_OBJECT_DICTIONARY", but its a variable-sized struct. Need to rebuild a slice from the length like is done for the array case.